### PR TITLE
Fix frontend unit tests

### PR DESCRIPTION
**Pull request by ramsays**
Working to get all of the proper modules listed in the `spec.ts` files to get unit tests up and running. Have a few fixed so far, still got some to go!

### CODE REVIEW

**Comment by ramsays** [comment]
@joshcosta I have commented out every failing test case, so now all we need to do is work through each one and make them work!

**Comment by ramsays** [comment]
I now have all the tests passing fairly consistently. See if y'all can reproduce them @vikaspotluri123 @joshcosta . Now it just all needs to be linted!

**Comment by ramsays** [comment]
I have all tests passing + lint. Please repro and verify it all works.

**Comment by vikaspotluri123** [comment]
I'll probably do this tonight!

**Comment by joshcosta** [comment]
Are we merging this with the tests still behaving unpredictably? I'm getting errors in the SemesterHomeComponent

File: src/app/gradetable/gradetable.component.ts [hunk 1]
```diff
@@ -140,13 +140,13 @@ export class GradetableComponent implements OnInit, OnDestroy {
...
-		const category = this.grades.find(el => {
+		const category = this.grades.find(function(el) {
 			return (el.id === id);
 		});
 
-		if (!category.name) {
+		if (!category.name || !category.weight) {
 			await this.dialog.open(PopupComponent, {
-				data: {title: '', text: 'Please name your assignment before making it a category.', button: 'Ok'}
+				data: {title: '', text: 'Please provide a name and weight before creating a category.', button: 'Ok'}
```
Comment (vikaspotluri123) [inline src/app/gradetable/gradetable.component.ts]: @joshcosta can you check that this logic is still correct? I think it is

Comment (joshcosta) [reply src/app/gradetable/gradetable.component.ts]: Yes, the incoming is right

**Pull request merged**